dashboard/src/shared/Namespace.ts

```diff
@@ -50,3 +51,18 @@ export default class Namespace {
 export const definedNamespaces = {
   all: "_all",
 };
+
+export function getCurrentNamespace(currentNS: string, availableNS: string[]) {
+  if (currentNS) {
+    // If a namespace has been already selected, use it
+    return currentNS;
+  }
+  // Try to get a namespace from the auth token
+  const tokenNS = Auth.defaultNamespaceFromToken(Auth.getAuthToken() || "");
+  if (tokenNS) {
+    // Return the default namespace in the token
+    return tokenNS;
```

Comment:
We should validate that this ns is in `availableNS` - we had a recent report because in Edukates the token is from a different namespace to which the user has no access.

Reply:
good point.